src/app/components/PropertyMap.tsx

```diff
@@ -1,15 +1,221 @@
-import React, { FC, useEffect, useState } from "react";
-import {  Popup, Map as MapboxMap, NavigationControl, FullscreenControl } from "mapbox-gl";
+import React, { FC, useEffect, useState, useRef } from "react";
+import {  Map as MapboxMap, VectorSource, SourceVectorLayer } from "mapbox-gl";
 import { mapboxAccessToken, apiBaseUrl } from "../../config/config";
 import ZoomModal from "./ZoomModal";
 import { useFilter } from "@/context/FilterContext";
 import LegendControl from 'mapboxgl-legend';
 import 'mapboxgl-legend/dist/style.css';
 import '../globals.css';
+import Map, { 
+  Source, 
+  Layer, 
+  Popup,
+  NavigationControl, 
+  FullscreenControl, 
+  ScaleControl,
+  GeolocateControl
+} from 'react-map-gl';
+import type { FillLayer, CustomSource, VectorSourceRaw } from 'react-map-gl';
 
+//let popup: Popup | null = null;
 
-let popup: Popup | null = null;
+const minZoom = 4;
 
+const VectorTiles: VectorSourceRaw = {
+  id: "vacant_properties",
+  type: "vector",
+  tiles: [`${apiBaseUrl}/api/generateTiles/{z}/{x}/{y}`],
+  minzoom: minZoom,
+  maxzoom: 23
+}
+
+const layerStyle: FillLayer = {
+  id: "vacant_properties",
+  type: "fill",
+  source: "vacant_properties",
+  "source-layer": "vacant_properties",
+  paint: {
+    "fill-color": [
+      "match",
+      ["get", "guncrime_density"], // get the value of the guncrime_density property
+      "Bottom 50%", "#B0E57C", // Light Green
+      "Top 50%", "#FFD700", // Gold
+      "Top 25%", "#FF8C00", // Dark Orange
+      "Top 10%", "#FF4500", // Orange Red
+      "Top 5%", "#B22222", // FireBrick
+      "Top 1%", "#8B0000", // Dark Rednp
+      "#0000FF" // default color if none of the categories match
+    ],
+    "fill-opacity": 0.7
+  },
+  metadata: {
+    name: 'Guncrime Density',
+  }
+};
+
+
+
+function MapControls() {
+  return (
+    <>
+      <GeolocateControl position="top-left" />
+      <FullscreenControl position="top-left" />
+      <NavigationControl position="top-left" />
+      <ScaleControl />
+    </>
+  );
+}
+
+interface MapSourceAndLayerProps {
+  onClick: (event: MapEvent) => void;
+}
+
+const MapSourceAndLayer: React.FC<MapSourceAndLayerProps> = ({ onClick }) => {
+  return (
+    <Source id="vacant_properties" {...VectorTiles}>
+      <Layer {...layerStyle} onClick={onClick} />
+    </Source>
+  );
+};
+
+
+export default function PropertyMap() {
+  const [isZoomModalHidden, setIsZoomModalHidden] = useState(true);
+  const { filter } = useFilter();
+  const [map, setMap] = useState<MapboxMap | null>(null);
+  const legend = new LegendControl();
+  const [popupInfo, setPopupInfo] = useState<any>(null);
```

Comment:
we should define a type for popup, but since the popup template might change we can leave this for now